### ISSUE TITLE
Better autosave

### ISSUE THREAD

**Issue by Morgandri1** [zed-industries/zed#7680]
### Check for existing issues

- [X] Completed

### Describe the feature

I'd love support for autosave on key up/down if possible; whatever the lowest level of autosave is, i'd love to have it!

### If applicable, add mockups / screenshots to help present your vision of the feature

just... add an "always" option to the selector or something

**Comment by jochenleidner** [comment]
Agreed. I would live to eventually migrate from Sublime totally to Zed (okay, I will probably always need Emacs for macros or occasional ELISP). Sublime never loses anything, and it would be a big win to have that feature for ZED also.

**Comment by paolodelfino** [comment]
Maybe you mean this:
```
"autosave": {
    "after_delay": {
      "milliseconds": 0
    }
  },
  ```

**Comment by Morgandri1** [comment]
@paolodelfino I prefer VSCode's autosave function which is more efficient by waiting for keystrokes to stop. Polling as often as possible is <presumably> inefficient and i'd prefer to avoid it

**Comment by dragove** [comment]
I'm a vim user and I would like to save when leaving insert mode. What I configured in my neovim is like following:
```lua
autocmd({ "InsertLeave", "FocusLost", "BufLeave" }, {
  desc = "Autosave on some actions",
  command = "silent! wall",
  nested = true,
})
```

**Comment by JosephTLyons** [comment]
> [@paolodelfino](https://github.com/paolodelfino) I prefer VSCode's autosave function which is more efficient by waiting for keystrokes to stop. Polling as often as possible is inefficient and i'd prefer to avoid it

> I'm a vim user and I would like to save when leaving insert mode. What I configured in my neovim is like following:
> 
> autocmd({ "InsertLeave", "FocusLost", "BufLeave" }, {
>   desc = "Autosave on some actions",
>   command = "silent! wall",
>   nested = true,
> })

These options make a lot of sense to me, especially when working with slow/heavy language servers that re-trigger on file save.
